azurestack/resource_arm_virtual_network_gateway.go

```diff
@@ -0,0 +1,776 @@
+		}
+
+		if err := d.Set("ip_configuration", flattenArmVirtualNetworkGatewayIPConfigurations(gw.IPConfigurations)); err != nil {
+			return fmt.Errorf("Error setting `ip_configuration`: %+v", err)
+		}
+
+		vpnConfigFlat := flattenArmVirtualNetworkGatewayVpnClientConfig(gw.VpnClientConfiguration)
+		if err := d.Set("vpn_client_configuration", vpnConfigFlat); err != nil {
+			return fmt.Errorf("Error setting `vpn_client_configuration`: %+v", err)
+		}
+
+		bgpSettingsFlat := flattenArmVirtualNetworkGatewayBgpSettings(gw.BgpSettings)
+		if err := d.Set("bgp_settings", bgpSettingsFlat); err != nil {
+			return fmt.Errorf("Error setting `bgp_settings`: %+v", err)
+		}
+
+	}
+
+	flattenAndSetTags(d, &resp.Tags)
+
+	return nil
+}
+
+func resourceArmVirtualNetworkGatewayDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).vnetGatewayClient
+	ctx := meta.(*ArmClient).StopContext
+
+	resGroup, name, err := resourceGroupAndVirtualNetworkGatewayFromId(d.Id())
+	if err != nil {
+		return err
+	}
+
+	future, err := client.Delete(ctx, resGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error deleting Virtual Network Gateway %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for deletion of Virtual Network Gateway %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	return nil
+}
+
+func getArmVirtualNetworkGatewayProperties(d *schema.ResourceData) (*network.VirtualNetworkGatewayPropertiesFormat, error) {
+	gatewayType := network.VirtualNetworkGatewayType(d.Get("type").(string))
+	vpnType := network.VpnType(d.Get("vpn_type").(string))
+	enableBgp := d.Get("enable_bgp").(bool)
+
+	// ActiveActive is not supported by AzureStack
+	// activeActive := d.Get("active_active").(bool)
+
+	props := &network.VirtualNetworkGatewayPropertiesFormat{
+		GatewayType: gatewayType,
+		VpnType:     vpnType,
+		EnableBgp:   &enableBgp,
+
+		// ActiveActive is not supported by AzureStack
+		// ActiveActive:     &activeActive,
+
+		Sku:              expandArmVirtualNetworkGatewaySku(d),
+		IPConfigurations: expandArmVirtualNetworkGatewayIPConfigurations(d),
+	}
+
+	if gatewayDefaultSiteID := d.Get("default_local_network_gateway_id").(string); gatewayDefaultSiteID != "" {
+		props.GatewayDefaultSite = &network.SubResource{
+			ID: &gatewayDefaultSiteID,
+		}
+	}
+
+	if _, ok := d.GetOk("vpn_client_configuration"); ok {
+		props.VpnClientConfiguration = expandArmVirtualNetworkGatewayVpnClientConfig(d)
+	}
+
+	if _, ok := d.GetOk("bgp_settings"); ok {
+		props.BgpSettings = expandArmVirtualNetworkGatewayBgpSettings(d)
+	}
+
+	// Sku validation for policy-based VPN gateways
+	if props.GatewayType == network.VirtualNetworkGatewayTypeVpn && props.VpnType == network.PolicyBased {
+		ok, err := evaluateSchemaValidateFunc(string(props.Sku.Name), "sku", validateArmVirtualNetworkGatewayPolicyBasedVpnSku())
```

Comment:
This could be simplified to:
```go
if ok, err := evaluateSchemaValidateFunc(string(props.Sku.Name), "sku", validateArmVirtualNetworkGatewayPolicyBasedVpnSku()); !ok {
			return nil, err
		}
```